### PR TITLE
Constructor injection for ViewModels

### PR DESCRIPTION
**Pull request by denis-tsv**

### CODE REVIEW

File: samples/MvvmSample.Core/ViewModels/AsyncRelayCommandPageViewModel.cs [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
 using System.Threading.Tasks;
 using CommunityToolkit.Mvvm.Input;
+using MvvmSample.Core.Services;
 
 namespace MvvmSample.Core.ViewModels;
 
 public class AsyncRelayCommandPageViewModel : SamplePageViewModel
 {
-    public AsyncRelayCommandPageViewModel()
+    public AsyncRelayCommandPageViewModel(IFilesService filesService) : base(filesService)
```
Comment (Sergio0694) [inline samples/MvvmSample.Core/ViewModels/AsyncRelayCommandPageViewModel.cs:13]: Nit: all these `base` calls should be in a new line.
```suggestion
    public AsyncRelayCommandPageViewModel(IFilesService filesService)
        : base(filesService)
```

File: samples/MvvmSampleUwp/Views/AsyncRelayCommandPage.xaml.cs [hunk 2]
```diff
@@ -14,5 +16,11 @@ public sealed partial class AsyncRelayCommandPage : Page
     public AsyncRelayCommandPage()
     {
         this.InitializeComponent();
+
+        ViewModel = Ioc.Default.GetRequiredService<AsyncRelayCommandPageViewModel>();
+
+        DataContext = ViewModel;
     }
+
+    public AsyncRelayCommandPageViewModel ViewModel { get; }
```
Comment (Sergio0694) [inline samples/MvvmSampleUwp/Views/AsyncRelayCommandPage.xaml.cs:25]: We can avoid the second state, all these should just be cases over `DataContext`.
As in:
```csharp
// Constructor
DataContext = Ioc.Default.GetRequiredService<AsyncRelayCommandPageViewModel>();

// Property
public AsyncRelayCommandPageViewModel ViewModel => (AsyncRelayCommandPageViewModel)DataContext;
```

**Review by Sergio0694 — CHANGES_REQUESTED** [review]
Looks good!
Just a couple comments 😄

File: samples/MvvmSample.Core/ViewModels/AsyncRelayCommandPageViewModel.cs (hunk 1 above)
Comment (denis-tsv) [reply samples/MvvmSample.Core/ViewModels/AsyncRelayCommandPageViewModel.cs]: fixed

File: samples/MvvmSampleUwp/App.xaml.cs [hunk 3]
```diff
@@ -39,14 +41,24 @@ protected override void OnLaunched(LaunchActivatedEventArgs e)
...
-            // Register services
-            Ioc.Default.ConfigureServices(
-                new ServiceCollection()
-                .AddSingleton<IFilesService, FilesService>()
-                .AddSingleton<ISettingsService, SettingsService>()
-                .AddSingleton(RestService.For<IRedditService>("https://www.reddit.com/"))
-                .BuildServiceProvider());
-        }
+                // Register services
+                Ioc.Default.ConfigureServices(
```
Comment (Sergio0694) [inline samples/MvvmSampleUwp/App.xaml.cs:45]: Just noticed: looks like this accidentally got indented 4 spaces too much?
Same for the closing bracket at the end 🤔

**Review by Sergio0694 — CHANGES_REQUESTED** [review]

File: samples/MvvmSampleUwp/Views/AsyncRelayCommandPage.xaml.cs (hunk 2 above)
Comment (denis-tsv) [reply samples/MvvmSampleUwp/Views/AsyncRelayCommandPage.xaml.cs]: fixed

File: samples/MvvmSampleUwp/App.xaml.cs (hunk 3 above)
Comment (denis-tsv) [reply samples/MvvmSampleUwp/App.xaml.cs]: merge issue ) fixed

**Review by Sergio0694 — APPROVED** [review]
Looks great, thank you! 🙌

**Pull request merged**